### PR TITLE
fix executing multiple SQL statements when using DuckDB with autopandas

### PR DESCRIPTION
**Pull request by edublancas**
## Issue number

Closes https://github.com/ploomber/jupysql/issues/674

## Checklist before requesting a review

- [ ] Performed a self-review of my code
- [ ] Formatted my code with [`pkgmt format`](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#linting-formatting)
- [ ] Added [tests](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#testing) (when necessary).
- [ ] Added [docstring](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#documenting-changes-and-new-features) documentation and update the [changelog](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#changelog) (when needed)

----
:books: Documentation preview :books:: https://jupysql--676.org.readthedocs.build/en/676/

### CODE REVIEW

File: src/tests/integration/test_duckDB.py [hunk 1]
```diff
@@ -27,3 +27,12 @@ def test_auto_commit_mode_off(ip_with_duckDB, caplog):
     # Check the tables is created
     tables_out = ip_with_duckDB.run_cell("%sql SHOW TABLES;").result
     assert any("weather" == table[0] for table in tables_out)
+
+
+def test_autopandas_with_multiple_statements(ip_empty):
+    ip_empty.run_cell("%sql duckdb://")
+    # ip_empty.run_cell("%config SqlMagic.autopandas=True")
```
Comment (neelasha23) [inline src/tests/integration/test_duckDB.py:34]: This is commented. I think we should add for both `autopandas=True/False`

File: src/tests/integration/test_duckDB.py [hunk 2]
```diff
@@ -27,3 +27,12 @@ def test_auto_commit_mode_off(ip_with_duckDB, caplog):
...
+    ip_empty.run_cell("%sql CREATE TABLE weather (city VARCHAR,);")
+    out = ip_empty.run_cell("%sql SELECT * FROM weather; SELECT * FROM weather;")
```
Comment (neelasha23) [inline src/tests/integration/test_duckDB.py:36]: can we try with two different SQL statements?

File: src/tests/integration/test_duckDB.py [hunk 3]
```diff
@@ -27,3 +27,12 @@ def test_auto_commit_mode_off(ip_with_duckDB, caplog):
...
+
+    assert out.error_in_exec is None
```
Comment (neelasha23) [inline src/tests/integration/test_duckDB.py:38]: any way to test the `out.result` also?

File: CHANGELOG.md [hunk 4]
```diff
@@ -8,6 +8,7 @@
 * [Doc] Document --persist-replace in API section (#539)
 * [Fix] Fixed CI issue by updating `invalid_connection_string_duckdb` in `test_magic.py` (#631)
 * [Fix] Refactored `ResultSet` to lazy loading (#470)
+* [Fix] Error when executing multiple SQL statements when using DuckDB with `autopandas` on
```
Comment (yafimvo) [inline CHANGELOG.md:11]: we don't have the issue number here (674)

File: src/sql/run.py [hunk 5]
```diff
@@ -590,9 +590,13 @@ def run(conn, sql, config):
         # returning only when sql is empty string
         return "Connected: %s" % conn.name
 
-    for statement in sqlparse.split(sql):
-        first_word = sql.strip().split()[0].lower()
+    statements = sqlparse.split(sql)
+    last_statement_is_select = _first_word(statements[-1])
```
Comment (yafimvo) [inline src/sql/run.py:594]: How do we know that the last statement is "select"?

**Review by yafimvo — COMMENTED** [review]
lgtm. added some comments and also please rebase.

**Review by tonykploomber — APPROVED** [review]

**Pull request closed without merging**